src/ScheduledDiagnostics.jl

```diff
@@ -158,7 +158,7 @@ function ScheduledDiagnostic(;
     reduction_time_func = nothing,
     compute_schedule_func = EveryStepSchedule(),
     output_schedule_func = isnothing(reduction_time_func) ?
-                           compute_schedule_func : EveryStepSchedule(),
+                           deepcopy(compute_schedule_func) : EveryStepSchedule(),
```

Comment:
Why would this need a deep copy?

Reply:
Because in general `compute_schedule_func` contains a mutable object that is used to track when the it was last called

Reply:
Doesn't that mean that the deep copy can become out of sync from what is passed in (if the input has any mutable data)?

Reply:
Yes, that's exactly the problem being fixed here.

Suppose you have a struct that holds the last time the callback was called: if you use this struct for both compute and output, calling `compute` will update the last time the callback was called, so that `output` will never happen. 

With a deepcopy, we allow them to be out of sync, so that they can happen independently.

Let me also note this only changes the default constructor that is currently never used.